### DISCUSSION TITLE
the random attribute in the formula fields is not persistent

### DISCUSSION THREAD

**Discussion by oswaldB** [Budibase/budibase#2006] [Help] [answered]
Hello, 

The random attribute in the formula {{ random 5 5000 }} regenerate a number every time and is not persistent.

Is it normal?

Thank you

**Comment by mike12345567** [comment, marked as answer]
Hi @oswaldB yes this just uses the javascript `Math.random()` under the hood, which will just generate a new number everytime it is called.

**Reply by oswaldB** [reply]
ok thank you.

Any idea on how to create a persistant one? using the automation?

**Reply by oswaldB** [reply]
and thank you for your quick reply :)

**Reply by PGLange** [reply]
Hm, one approach could be to save it (generated with automatisation -> CMD, use python here) to table called parameters, with "persistent settings"... Use a dataProvider to access these fields... This isn't even hacky :) 
Hope this may help....

**Reply by PGLange** [reply]
or with batch itself
https://linuxconfig.org/generating-random-numbers-in-bash-with-examples

**Reply by oswaldB** [reply]
I tried to use another approach in automation:
+ on row created event I add an update row action and generate a random number in a text column. => not working.

I can't even update a row when a new row is created. Do you have any documentation on that point?

**Reply by mike12345567** [reply]
Hi @oswaldB - I had a go at this myself, there appears to be an issue with the currently released version, some changes after the SQL update, I've put a PR up to fix these, which should be released ASAP.

https://github.com/Budibase/budibase/pull/2012

**Comment by oswaldB** [comment]
Oki thanks a lot.

**Reply by oswaldB** [reply]
Hello

In my case it is not working. Have you released the fix, yet?

Thank you

**Reply by mike12345567** [reply]
Hi @oswaldB - which version of Budibase are you running? Were you able to update? Release `0.9.77` contains the fix.

**Reply by oswaldB** [reply]
I am on the 0.9.77

**Reply by oswaldB** [reply]
![image](https://user-images.githubusercontent.com/3023526/126150638-a13b22e1-9220-4828-b649-c7fc46e9098c.png)

**Reply by oswaldB** [reply]
the Hello is never there.

**Reply by oswaldB** [reply]
I tried to add a row from the data tabs and from the form in the publish app

**Reply by mike12345567** [reply]
Hi @oswaldB - just noticed in your picture, you need to activate your automation, in the top right there is a little pause and play symbol (can see it in screenshot) you need to click the play button to active the automation!

Sorry this is a bit confusing a number of people have had an issue with this, I'll raise an issue to make this more clear.

**Reply by oswaldB** [reply]
Thank you it is working fine!

It could be a good idea to make it more clear ;)
